File: 128-longest-consecutive-sequence/longest-consecutive-sequence.dart
class Solution {
  List<int> getFirsts(List<int> nums){
    final numsSet = Set.from(nums);
    final firsts = <int>[];
    for (int item in numsSet){
        if(!numsSet.contains(item-1)){
            firsts.add(item);
        }
    }
    return firsts;
  }


  int longestConsecutive(List<int> nums) {
    final firsts = getFirsts(nums);
    final numsSet = Set.from(nums);
    int longest = 0;

    for (int val in firsts){
        int counter = 1;
        while (numsSet.contains(val+1)){
            counter ++;
            val ++;
        }
        longest = max(longest, counter);
    }
    return longest;

  }
}


// [100,4,200,1,3,2,99]
// -> 1 2 3 4
// -> 99 100
// -> 200

// Firsts, Zincir baslangiclarini bul -> 1 , 99 , 200

// firsts listesi loop cevir, enUzun degisteknini gguncelle

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        numSet = set(nums)
        longest = 0

        for n in nums:
            cur = n
            while (cur-1) in numSet:
                cur -= 1
            count = 0
            while (cur in numSet):
                numSet.remove(cur)
                count += 1
                cur += 1
            longest = max(longest, count)
        return longest
            

        